skrub/_dataframe/tests/test_common.py

```diff
@@ -165,13 +166,28 @@ def test_all_null_like(df_module):
         expected = expected.astype("bool")
     df_module.assert_column_equal(ns.is_null(col), expected)
 
+    col = ns.all_null_like(df_module.example_column, length=2)
+    assert ns.shape(col)[0] == 2
+
+    # Test can set length greater than column length
+    max_len = ns.shape(df_module.example_column)[0]
+    col = ns.all_null_like(df_module.example_column, length=max_len + 2)
+    assert ns.shape(col)[0] == (max_len + 2)
+
 
 def test_concat_horizontal(df_module, example_data_dict):
     df1 = df_module.make_dataframe(example_data_dict)
     df2 = ns.set_column_names(df1, list(map("{}1".format, ns.column_names(df1))))
     df = ns.concat_horizontal(df1, df2)
     assert ns.column_names(df) == ns.column_names(df1) + ns.column_names(df2)
 
+    # Test concatenating dataframes with the same column names
+    df2 = df1
+    df = ns.concat_horizontal(df1, df2)
+    assert ns.shape(df) == (4, 16)
+    for n in ns.column_names(df)[8:]:
+        assert re.match(r".*__skrub_[0-9a-f]+__", n)
+
```

Comment:
I realize we don't have a test for concat_horizontal for pandas dataframes that have different indexes (of the same length). it's important that we reset the indexes in concat_horizontal because by default pandas will align them. In this or a later PR it would be nice to add a test